pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go

```diff
@@ -71,7 +73,32 @@ func (admitter *SecretAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissio
 			if err != nil {
 				log.Info("Connection test failed, yet passing", "status", status, "error", err.Error())
 			} else {
-				log.Info("Test succeeded, passing")
+				// Perform validation for the provided engine CA certificate,
+				// initiate dummy image transfer to make sure the certificate is correct and that the migration will work in a later stage.
+				if createdForProviderType == "ovirt" {
+					if isCaOK, err := ovirt.TestDownloadOvfStore(secret); isCaOK {
+						if err != nil {
+							return &admissionv1.AdmissionResponse{
+								Allowed: false,
+								Result: &metav1.Status{
+									Code:    http.StatusBadRequest,
+									Message: err.Error(),
+								},
+							}
+						}
+					}
+					log.Info("Test credentials and CA certificat succeeded, passing")
+				} else {
+					log.Info("Test credentials succeeded but engine CA certificate is not valid")
```

Comment:
but in this case, when it's not ovirt, `engine CA certificate` is irrelevant, right?

Reply:
yes, I've split the cases between ovirt and other provider's types, and also added good flow and bad flow specific to ovirt.

Reply:
positive and negative flows :)